ADR/user-specified-tokens-build.md

```diff
@@ -0,0 +1,28 @@
+# Enable users to configure tokens/credentials for a Build/Test Pipeline
+
+Date: 2023-03-03
+
+## Status
+
+Provisional
+
+
+## Context
+
+As a user, I would like to configure sensitive information/tokens/credentials for the build/test pipeline to consume as part of a partner task.
+
+
+## Decision
+
+1. UI will take in sensitive information as form input and upload it to SPI using the user's authentication token. See https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/docs/USER.md#uploading-access-token-to-spi-using-kubernetes-secret .
+2. In the above form, the user should be able to choose the name of the `Secret` from a dropdown. 
+The dropdown would be a pre-populated list of `Secret` names associated with known partner integrations. The user should also be allowed to provide a name barring the ones that already exist in the namespace.
```

Comment:
we need a naming convention to avoid the collision of secret names between partner tasks.

Reply:
or perhaps better a completely random name and some sensible labels?

Reply:
I think completely random variable names would be worse from a UI perspective than scoping variables to a specific task. Scoping variables to a task would also make it easier to configure the task in a Pipeline definition as the secret configuration would remain constant and wouldn't have to be fiddled with based on the secret set. Even if the "sensible labels" were used for displaying the dropdown instead of the name of the secret, debugging processes would likely be complicated if randome secret names were used instead.

Reply:
Updated the ADR:

* UI is responsible for the name of the `Secret`. UI has the freedom to choose legal names based on what's available on the namespace.
* UI sets a discoverable label which the build-service or integration service would depend on.

Reply:
* UI shows the user the list of parameter names that she could optionally set.